identity-account-storage/src/stronghold/snapshot.rs

```diff
@@ -44,25 +43,12 @@ impl Snapshot {
     &self.path
   }
 
-  pub fn vault<T>(&self, name: &T, flags: &[StrongholdFlags]) -> Vault<'_>
-  where
-    T: AsRef<[u8]> + ?Sized,
-  {
-    Vault::new(&self.path, name, flags)
-  }
-
-  pub fn store<T>(&self, name: &T, flags: &[StrongholdFlags]) -> Store<'_>
-  where
-    T: AsRef<[u8]> + ?Sized,
-  {
-    Store::new(&self.path, name, flags)
+  pub fn vault(&self, client_path: ClientPath) -> Vault<'_> {
+    Vault::new(&self.path, client_path, &[])
   }
 
-  pub fn records<T>(&self, name: &T, flags: &[StrongholdFlags]) -> Records<'_>
-  where
-    T: AsRef<[u8]> + ?Sized,
-  {
-    Records::new(&self.path, name, flags)
+  pub fn store(&self, client_path: ClientPath) -> Store<'_> {
+    Store::new(&self.path, client_path, &[])
```

Comment:
Would making the parameter generic with `T: Into<ClientPath>` be better or do we want to avoid generics since these are private structs anyway?

Reply:
That would work just as well, and that's what I had initially, but then changed it to be super explicit about what the client path is. Since stronghold just uses `Vec<u8>`, almost anything goes as the client identifier. I mainly did this to not lose my sanity while trying to make sure to consistently use a fixed string or a did as the client path, which is important for forward compatibility. So I'll leave it as more explicit for now.